src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3676,7 +3676,7 @@ MethodSymbol addPrintMembersMethod()
                 MethodSymbol printMembersMethod;
                 if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingPrintMembersMethod))
                 {
-                    printMembersMethod = new SynthesizedRecordPrintMembers(this, memberOffset: members.Count, diagnostics);
+                    printMembersMethod = new SynthesizedRecordPrintMembers(this, generatedMembers: members, userDefinedMembers: userDefinedMembers, diagnostics);
```

Comment:
The reader may be unsettled by a mutable collection being passed to a symbol constructor here. I can see in the implementation that it's not mutated or retained by the symbol, but I think it will probably be best to go back to passing only the `members.Count` here and relying only on the `userDefinedMembers` to determine `isReadOnly`.

To verify our assumption, let's add an assert before `members.AddRange(membersSoFar)` to the effect of `Debug.Assert(isRecordClass || !members.Any(m is PropertySymbol { GetMethod.IsEffectivelyReadOnly: false })`. This would verify that on a record struct, we *never* synthesize a non-`readonly` property.

Reply:
Yes, this would be better. One more reason for that is that the current approach could skip properties generated after `addPrintMembersMethod`. Updated PR, thank you.